### ISSUE TITLE
Support additional color definition formats

### ISSUE THREAD

**Issue by BVollmerhaus** [jaagr/polybar#1296]
Currently Polybar only supports color definitions in RGB hex triplets. It would be nice to have additional, more human-readable formats such has HSL or RGB. Especially HSL would allow for easier fine-tuning of colors.

**Comment by patrick96** [comment]
Good idea, I'm currently rewriting the config parser. I may do a reworking of different value types after that. 
If you want to use hsl right now, you could use a converter like [this one](http://serennu.com/colour/hsltorgb.php).

**Comment by Sircular** [comment]
Additionally, it would be nice if it were possible to define colors based on other colors. I use pywal, which changes the xresources colors, and it would be nice to be able to say "20% darker than color4" for example.

**Comment by patrick96** [comment]
@Sircular once we have a clean implementation of different color definitions, I think we could more or less easily implement this.

**Comment by NBonaparte** [comment]
Hopefully no one will request the CIELAB color space...
